addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -1017,21 +1017,43 @@ def reserve_pallet(self, pallet_name):
         Raises a ValidationError if the pallet is already reserved for another
         batch.
         """
+        Picking = self.env['stock.picking']
         PickingBatch = self.env['stock.picking.batch']
 
         self.ensure_one()
 
-        conflicting_batch = PickingBatch.search([
-            ('id', '!=', self.id),
-            ('state', '=', 'in_progress'),
-            ('u_last_reserved_pallet_name', '=', pallet_name),
-        ])
+        reserve_pallet_per_picking = self.picking_ids.filtered('picking_type_id.u_reserve_pallet_per_picking')
```

Comment:
If this is true we should check we have `picking != None`.

Batch has a field u_picking_type_ids or similar from which you can get `u_reserve_pallet_per_picking` with a shorter code.